### DISCUSSION TITLE
How to make validator case insensitive?

### DISCUSSION THREAD

**Discussion by durianwaffle** [eclipse-langium/langium#1706] [Q&A] [answered]
I found that if I defined a function to have output parameters in upper case. When I access its parameter in lower case. It will show validation error "undefined variable".

How to fix this issue?

Currently if we put function parameter definitions in uppercase in LSP, it can only recognize function parameter definitions in uppercase when user use it

**Comment by cdietrich** [comment, marked as answer]
hmmmm. have you customized error message. normal error message would be "could not resolve reference to Variable named 'x'"?
if you want to do case insensitive scoping check out 
https://langium.org/docs/recipes/lexing/case-insensitive-languages/#case-insensitivity-for-identifiers-and-cross-references

**Reply by durianwaffle** [reply]
Yeah I customised the message. 
Thanks for the reference of case insensitive scope checking!
